packages/engine/src/postprocessing/components/PostProcessingComponent.tsx

```diff
@@ -0,0 +1,572 @@
+      }
+    }
+
+    //always have the smaa effect
+    const smaaPreset = getState(RenderSettingsState).smaaPreset
+    const smaaEffect = new SMAAEffect({
+      preset: smaaPreset,
+      edgeDetectionMode: EdgeDetectionMode.COLOR
+    })
+
+    //always have the outline effect for the highlight selection
+    const outlineEffect = new OutlineEffect(scene.value, camera.value, getState(HighlightState))
+    outlineEffect.selectionLayer = ObjectLayers.HighlightEffect
+    effectsVal[Effects.OutlineEffect] = outlineEffect
+    renderer.value.effectComposer[Effects.OutlineEffect] = outlineEffect
+
+    if (renderer.value.effectComposer.EffectPass) {
+      renderer.value.effectComposer.removePass(renderer.value.effectComposer.EffectPass)
+    }
+
+    const effectArray = Object.values(effectsVal)
+    renderer.value.effectComposer.EffectPass = new EffectPass(camera.value, ...effectArray)
+    renderer.value.effectComposer.addPass(renderer.value.effectComposer.EffectPass)
+  }, [effects])
+
+  useEffect(() => {
+    if (!rendererEntity) return
+
+    if (getState(EngineState).isEditor) changeRenderMode()
+  }, [rendererEntity, postprocessingComponent.enabled, postprocessingComponent.effects, renderMode])
+
+  return (
+    <>
+      {Object.entries(postprocessingComponent.effects.value).map(([effectKey, schema], index) => {
+        return (
+          <EffectReactor
+            key={index}
+            entity={entity}
+            effectKey={effectKey}
+            schema={schema}
+            effects={effects}
+            camera={camera}
+            uiEffects={postprocessingComponent.effects}
+            scene={scene}
+            lut1DEffectTexture={lut1DEffectTexture}
+            lut3DEffectTexture={lut3DEffectTexture}
+            velocityDepthNormalPass={velocityDepthNormalPass}
+            useVelocityDepthNormalPass={useVelocityDepthNormalPass}
+            normalPass={normalPass}
+            depthDownsamplingPass={depthDownsamplingPass}
+            composer={renderer.value.effectComposer}
+          ></EffectReactor>
+        )
+      })}
+    </>
+  )
+}
+
+const EffectReactor = (props: {
+  entity: Entity
+  effectKey: string
+  schema: any
+  effects: State<Record<string, Effect>, {}>
+  camera: State<ArrayCamera>
+  uiEffects: State<EffectPropsSchema, {}>
+  scene: State<Scene, {}>
+  lut1DEffectTexture: Texture | null
+  lut3DEffectTexture: Texture | null
+  velocityDepthNormalPass: any
+  useVelocityDepthNormalPass: StateMethods<boolean, {}>
+  normalPass: State<CustomNormalPass, {}>
+  depthDownsamplingPass: State<DepthDownsamplingPass, {}>
+  composer: EffectComposer
+}) => {
+  const {
+    entity,
+    effectKey,
+    schema,
+    effects,
+    camera,
+    uiEffects,
+    scene,
+    lut1DEffectTexture,
+    lut3DEffectTexture,
+    velocityDepthNormalPass,
+    useVelocityDepthNormalPass,
+    normalPass,
+    depthDownsamplingPass,
+    composer
+  } = props
+  const lookUp = {
```

Comment:
this mapping can be a hyperflux state or at least a module scope object (like the ones in PostProcessing.ts) such that it can be extensible